rfcs/text/0000-image-base-transcriptomics.md

```diff
@@ -0,0 +1,497 @@
+error, and the IntensityTable supports analysis and QC for this
+process.
+
+##### [SegmentationMask]
+
+In order to assign the spots identified above to individual cells, the
+locations of cells within the images must be identified. Compressed
+segmentation masks are a compact way to represent these data, which
+store, for each cell, the portion of the image that is attributed to
+that cell.
+
+##### [ExpressionMatrix]
+
+*starfish* combines spot calls from the IntensityTable with cell areas
+in the SegmentationMask to produce expression matrices. These are saved
+in zarr format and look identical to the matrices emitted by the Data
+Analysis Service pipelines, with the exception that each cell is
+additionally annotated with metadata that specify the x, y, and z
+location of the data.
+
+##### Processed Images
+
+Image-based transcriptomics experiments suffer from several sources of
+error and variation which are removed via image processing. Microscope
+stages or tissue slices may shift as reagents are added and removed
+across imaging rounds, or which spectral filters are swapped between
+channels. In addition, reagents can non-specifically bind to cellular
+material, causing autofluorescence. Prior to spot-calling, images can be
+aligned, background removed, and spots computationally enhanced.
+*starfish* can output these processed images, annotated with information
+sufficient to re-do the processing, so that the images input to spot
+calling can be evaluated by users. If emitted, processed images will be
+generated in SpaceTx Format.
+
+#### HCA Sample Metadata
+
+The HCA metadata team has built a metadata specification draft for
+imaging data that references data in SpaceTx-Format. The issue that
+formed the draft can be found
+[<span class="underline">here</span>](https://github.com/HumanCellAtlas/metadata-schema/issues/368).
+
+### Data and Metadata validation
+
+#### Input Data validation
+
+SpaceTx Format consists of a json file that specifies how to compose a
+dataset from a set of two-dimensional images. As a result, it is
+possible to rapidly validate that a file is complete and internally
+consistent. In addition, SpaceTx Format stores per-chunk checksum
+information, enabling the data to be checked for integrity. It is
+possible to carry out additional, more data-specific checks, such as
+that images are not blank, contain spots, or contain cells, however
+these checks take increasing amounts of time.
+
+OME and SpaceTx can provide a docker container that validates image
+datasets with the highest level of stringency possible in a DCP-provided
+"maximum validation time".
+
+#### Output Data validation
+
+Each of the output formats produced by SpaceTx can be stored in Zarr and
+read (on-demand) by *starfish* and xarray. SpaceTx provides a docker
+container that loads and validates each output file using the xarray
+library. Validation can be tuned to occur within a DCP-provided target
+validation time.
```

Comment:
What do you mean by "can be tuned to occur within a ...time"?

Reply:
If we told "validation can cost $1 or take 30s" starfish will validate the json files. If it can cost many dollars and take hours, we can validate the image data. 

If something in between, we could randomly test a few images and all the json files.

Reply:
If we told "validation can cost $1 or take 30s" starfish will validate the json files. If it can cost many dollars and take hours, we can validate the image data. 

If something in between, we could randomly test a few images and all the json files.

Reply:
ok, thanks for the clarification.